### PR TITLE
#370 Directory's organization and organizationMapping

### PR DESCRIPTION
**Pull request by maranai**

### CODE REVIEW

**Comment by mrioan** [comment]
@maranai 
1.Just simply change `@version` annotations to mach the proper release version. 
2. Please also be sure to have CI passing. 

Approved once those two things are met. Thanks

**Comment by dogeared** [comment]
Need to update javadocs

File: api/src/main/java/com/stormpath/sdk/directory/Directory.java [hunk 1]
```diff
@@ -381,4 +385,156 @@
      */
     Directory saveWithResponseOptions(DirectoryOptions responseOptions);
 
-}
+    /**
+     * Returns a paginated list of all {@link com.stormpath.sdk.organization.Organization}s that have this Directory as
```
Comment (lhazlewood) [inline api/src/main/java/com/stormpath/sdk/directory/Directory.java]: Minor copy edit suggestion:

> Returns a paginated list of all {@link Organization}s that map this directory as an account store.  For any Organizations returned, this directory and all of the accounts in it are seen as part of the total account population attributed to the respective Organization.

File: api/src/main/java/com/stormpath/sdk/directory/Directory.java [hunk 2]
```diff
@@ -381,4 +385,156 @@
...
+     * an {@link com.stormpath.sdk.directory.AccountStore}.
+     * @see #getOrganizations(java.util.Map)
+     * @see #getOrganizationAccountStoreMappings()
+     *
+     * @since 1.0.RC7.7
+     */
+    OrganizationList getOrganizations();
+
+    /**
+     * Returns a paginated list of the {@link com.stormpath.sdk.organization.Organization}s that have this Directory as
```
Comment (lhazlewood) [inline api/src/main/java/com/stormpath/sdk/directory/Directory.java]: > Returns a paginated list of all {@link Organization}s that map this directory as an account store and also match the specified query criteria.  For any Organizations returned, this directory and all of the accounts in it are seen as part of the total account population attributed to the respective Organization.

File: api/src/main/java/com/stormpath/sdk/directory/Directory.java [hunk 3]
```diff
@@ -381,4 +385,156 @@
...
+     * @return a paginated list of the {@link com.stormpath.sdk.organization.Organization}s that have this Directory as
+     * an {@link com.stormpath.sdk.directory.AccountStore} that match the specified query criteria.
+     * @see #getOrganizationAccountStoreMappings(java.util.Map)
+     *
+     * @since 1.0.RC7.7
+     */
+    OrganizationList getOrganizations(Map<String, Object> queryParams);
+
+    /**
+     * Returns a paginated list of the {@link com.stormpath.sdk.organization.Organization}s that have this Directory as
```
Comment (lhazlewood) [inline api/src/main/java/com/stormpath/sdk/directory/Directory.java]: > Returns a paginated list of all {@link Organization}s that map this directory as an account store and also match the specified query criteria.  For any Organizations returned, this directory and all of the accounts in it are seen as part of the total account population attributed to the respective Organization.

File: api/src/main/java/com/stormpath/sdk/directory/Directory.java [hunk 4]
```diff
@@ -381,4 +385,156 @@
...
+     * @return a paginated list of the {@link com.stormpath.sdk.organization.Organization}s that have this Directory as
+     * an {@link com.stormpath.sdk.directory.AccountStore} that match the specified query criteria
+     * @see #getOrganizationAccountStoreMappings(com.stormpath.sdk.organization.OrganizationAccountStoreMappingCriteria).
+     *
+     * @since 1.0.RC7.7
+     */
+    OrganizationList getOrganizations(OrganizationCriteria criteria);
+
+    /**
+     * Returns a paginated list of all {@link com.stormpath.sdk.organization.OrganizationAccountStoreMapping}s that have
```
Comment (lhazlewood) [inline api/src/main/java/com/stormpath/sdk/directory/Directory.java]: > Returns a paginated list of all {@link OrganizationAccountStoreMapping}s that represent the link between an organization and this directory.  For any Organization reflected in a mapping, this directory and all of the accounts in it are seen as part of the total account population attributed to the respective Organization.

File: api/src/main/java/com/stormpath/sdk/directory/Directory.java [hunk 5]
```diff
@@ -381,4 +385,156 @@
...
+     * this Directory as an {@link com.stormpath.sdk.directory.AccountStore}.
+     * @see #getOrganizationAccountStoreMappings(java.util.Map)
+     * @see #getOrganizations()
+     *
+     * @since 1.0.RC7.7
+     */
+    OrganizationAccountStoreMappingList getOrganizationAccountStoreMappings();
+
+    /**
+     * Returns a paginated list of the {@link com.stormpath.sdk.organization.OrganizationAccountStoreMapping}s that have
```
Comment (lhazlewood) [inline api/src/main/java/com/stormpath/sdk/directory/Directory.java]: > Returns a paginated list of all {@link OrganizationAccountStoreMapping}s that represent the link between an organization and this directory that match the specified query criteria.  For any Organization reflected in a mapping, this directory and all of the accounts in it are seen as part of the total account population attributed to the respective Organization.

File: api/src/main/java/com/stormpath/sdk/directory/Directory.java [hunk 6]
```diff
@@ -381,4 +385,156 @@
...
+     * @return a paginated list of the {@link com.stormpath.sdk.organization.OrganizationAccountStoreMapping}s that have
+     * this Directory as an {@link com.stormpath.sdk.directory.AccountStore}.
+     * @see #getOrganizations(java.util.Map)
+     *
+     * @since 1.0.RC7.7
+     */
+    OrganizationAccountStoreMappingList getOrganizationAccountStoreMappings(Map<String, Object> queryParams);
+
+    /**
+     * Returns a paginated list of the {@link com.stormpath.sdk.organization.OrganizationAccountStoreMapping}s that have
```
Comment (lhazlewood) [inline api/src/main/java/com/stormpath/sdk/directory/Directory.java]: > Returns a paginated list of all {@link OrganizationAccountStoreMapping}s that represent the link between an organization and this directory that match the specified query criteria.  For any Organization reflected in a mapping, this directory and all of the accounts in it are seen as part of the total account population attributed to the respective Organization.

**Pull request merged**